streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/RackAwareTaskAssignor.java

```diff
@@ -185,4 +191,224 @@ public boolean validateClientRack() {
         }
         return true;
     }
+
+    private int getCost(final TaskId taskId, final UUID clientId, final boolean inCurrentAssignment, final boolean isStateful) {
```

Comment:
I don't think that `clientId` is a `UUID`, but it should be rather a `String`? Or is this supposed to be the `processId` ?

("client" is a somewhat overloaded term unfortunately...)

Reply:
There is a `isStateful` flag -- do we need it? My understanding is that costs (as we only consider input topic partitions) are independent of stateful/stateless?

Reply:
What does `inCurrentAssignment` exactly mean?

Reply:
Here I meant `processId` using the UUID.

Stateless tasks have 0 `non_overlap_cost` while stateful tasks have postive `non_overlap_cost`

`inCurrentAssignment` means the task is assigned to current client referred to by `clientId`. If not in current assignment, there's possible `non_overlap_cost`